packages/client/src/client4.ts

```diff
@@ -3812,17 +3812,17 @@ export default class Client4 {
         );
     }
 
-    requestCloudTrial = (email = '') => {
+    requestCloudTrial = (subscriptionId: string, email = '') => {
         return this.doFetchWithResponse<CloudCustomer>(
             `${this.getCloudRoute()}/request-trial`,
-            {method: 'put', body: JSON.stringify({email})},
+            {method: 'put', body: JSON.stringify({email, subscription_id: subscriptionId})},
```

Comment:
Normally I'd say let the server do the validation rather than trusting input from the client, but since CWS checks it anyways and mattermost-server is a middleman, I'm ok with this.

Reply:
Agreed, why are we sending the subscription id over to the server? Doesn't the server have it in a cache?

Reply:
when making the request we had available the subscriptionId while the handleRequest in the server was grabbing the subscriptionId from the user context https://github.com/mattermost/mattermost-server/pull/20352/files#diff-5b4abeaf1bd0792473cef52eb3e3022b373d62124be75c83676027f10d381b0aL146

Reply:
Makes sense, thanks for clarifying!